task/common/machine/script.go

```diff
@@ -60,12 +60,24 @@ sudo mv terraform-provider-iterative* /usr/bin/tpi
 sudo chmod u=rwx,g=rx,o=rx /usr/bin/tpi
 sudo chown root:root /usr/bin/tpi
 
-curl --remote-name https://downloads.rclone.org/rclone-current-linux-amd64.zip
-python3 -m zipfile -e rclone-current-linux-amd64.zip .
-sudo cp rclone-*-linux-amd64/rclone /usr/bin
-sudo chmod u=rwx,g=rx,o=rx /usr/bin/rclone
-sudo chown root:root /usr/bin/rclone
-rm --recursive rclone-*-linux-amd64*
+extract_here(){
+  if [[ -n "$(which unzip)" ]]; then
+    unzip "$1"
+  elif [[ -n "$(which python)" ]]; then
+    python -m zipfile -e "$1" .
+  else
+    python3 -m zipfile -e "$1" .
+  fi
+}
+
+if [[ -z "$(which rclone)" ]]; then
+  curl --remote-name https://downloads.rclone.org/rclone-current-linux-amd64.zip
+  extract_here rclone-current-linux-amd64.zip
+  sudo cp rclone-*-linux-amd64/rclone /usr/bin
+  sudo chmod u=rwx,g=rx,o=rx /usr/bin/rclone
+  sudo chown root:root /usr/bin/rclone
+  rm --recursive rclone-*-linux-amd64*
+fi
```

Comment:
```suggestion
extract_here(){
  if command -v unzip 2>&1 > /dev/null; then
    unzip "$1"
  elif command -v python3 2>&1 > /dev/null; then
    python3 -m zipfile -e "$1" .
  else
    python -m zipfile -e "$1" .
  fi
}

if ! command -v rclone 2>&1 > /dev/null; then
  curl --remote-name https://downloads.rclone.org/rclone-current-linux-amd64.zip
  extract_here rclone-current-linux-amd64.zip
  sudo cp rclone-*-linux-amd64/rclone /usr/bin
  sudo chmod u=rwx,g=rx,o=rx /usr/bin/rclone
  sudo chown root:root /usr/bin/rclone
  rm --recursive rclone-*-linux-amd64*
fi
```

See also: [`command`](https://pubs.opengroup.org/onlinepubs/007904975/utilities/command.html)

Reply:
sure, if this is more POSIXy...

Reply:
I don't see you very convinced. Let me add an example:

### Built–in `which`

```console
$ which which
which: shell built-in command
$ result="$(which unobtanium)"
$ printf "result: %s\n" "$result"
result: unobtanium not found
$ if test -n "$result"; then echo exists; else echo does not exist; fi
exists
```

### External `which`

```console
$ which /usr/bin/which
/usr/bin/which
$ result="$(/usr/bin/which unobtanium)"
$ printf "result: %s\n" "$result"
result: 
$ if test -n "$result"; then echo exists; else echo does not exist; fi
does not exist
```

### Portable `command`

```console
$ which command
command: shell built-in command
$ if command -v unobtanium 2>&1 > /dev/null; then echo exists; else echo does not exist; fi
does not exist
```

Running macOS 10.15.7 and `zsh`

Reply:
:no-mac:

Reply:
Did you mean `:no-zsh:` instead?